internal/impl/postgresql/input_postgrecdc.go

```diff
@@ -0,0 +1,562 @@
+	}
+
+	tables, err = conf.FieldStringList("tables")
+	if err != nil {
+		return nil, err
+	}
+
+	if checkpointLimit, err = conf.FieldInt("checkpoint_limit"); err != nil {
+		return nil, err
+	}
+
+	streamSnapshot, err = conf.FieldBool("stream_snapshot")
+	if err != nil {
+		return nil, err
+	}
+
+	streamUncomited, err = conf.FieldBool("stream_uncomited")
+	if err != nil {
+		return nil, err
+	}
+
+	decodingPlugin, err = conf.FieldString("decoding_plugin")
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotMemSafetyFactor, err = conf.FieldFloat("snapshot_memory_safety_factor")
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotBatchSize, err = conf.FieldInt("snapshot_batch_size")
+	if err != nil {
+		return nil, err
+	}
+
+	if pgConnOptions, err = conf.FieldString("pg_conn_options"); err != nil {
+		return nil, err
+	}
+
+	if pgConnOptions != "" {
+		pgConnOptions = "options=" + pgConnOptions
+	}
+
+	if batching, err = conf.FieldBatchPolicy("batching"); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	pgconnConfig := pgconn.Config{
+		Host:     dbHost,
+		Port:     uint16(dbPort),
+		Database: dbName,
+		User:     dbUser,
+		TLSConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+		Password: dbPassword,
+	}
+
+	if tlsSetting == "none" {
+		pgconnConfig.TLSConfig = nil
+	}
+
+	snapsotMetrics := mgr.Metrics().NewGauge("snapshot_progress", "table")
+	replicationLag := mgr.Metrics().NewGauge("replication_lag_bytes")
+	snapshotMessageRate := mgr.Metrics().NewGauge("snapshot_message_rate")
+	snapshotRateCounter := NewRateCounter()
+
+	i := &pgStreamInput{
+		dbConfig:                pgconnConfig,
+		streamSnapshot:          streamSnapshot,
+		snapshotMemSafetyFactor: snapshotMemSafetyFactor,
+		slotName:                dbSlotName,
+		schema:                  dbSchema,
+		pgConnRuntimeParam:      pgConnOptions,
+		tls:                     pglogicalstream.TLSVerify(tlsSetting),
+		tables:                  tables,
+		decodingPlugin:          decodingPlugin,
+		streamUncomited:         streamUncomited,
+		temporarySlot:           temporarySlot,
+		snapshotBatchSize:       snapshotBatchSize,
+		snapshotMessageRate:     snapshotMessageRate,
+		snapshotRateCounter:     snapshotRateCounter,
+		batching:                batching,
+		checkpointLimit:         checkpointLimit,
+		cMut:                    sync.Mutex{},
+		msgChan:                 make(chan asyncMessage),
+
+		mgr:             mgr,
+		logger:          mgr.Logger(),
+		metrics:         mgr.Metrics(),
+		snapshotMetrics: snapsotMetrics,
+		replicationLag:  replicationLag,
+		inTxState:       atomic.Bool{},
+		releaseTrxChan:  make(chan bool),
+	}
+
+	r, err := service.AutoRetryNacksBatchedToggled(conf, i)
+	if err != nil {
+		return nil, err
+	}
+
+	return conf.WrapBatchInputExtractTracingSpanMapping("pg_stream", r)
+}
+
+func init() {
+	rng, _ := codename.DefaultRNG()
+	randomSlotName = strings.ReplaceAll(codename.Generate(rng, 5), "-", "_")
```

Comment:
Can we do this in `newPgStreamInput`? Otherwise we will use the same slot name if this is running in streams mode or there is a `broker` input that is reading from multiple databases.